src/transformers/data/data_collator.py

```diff
@@ -328,7 +338,9 @@ def torch_call(self, features):
                 [self.label_pad_token_id] * (sequence_length - len(label)) + list(label) for label in labels
             ]
 
-        batch = {k: torch.tensor(v, dtype=torch.int64) for k, v in batch.items()}
+        batch = {
+            k: (torch.tensor(v, dtype=torch.int64) if not isinstance(v, torch.Tensor) else v) for k, v in batch.items()
+        }
```

Comment:
Only the `label_name` key needs to be converted here, so maybe do this in the test before directly?